src/components/LayerControl.vue

```diff
@@ -98,21 +122,54 @@ export default {
         'layer-control--active': this.active
       }
     },
-    variantsOptions () {
+    layerVariantOptions () {
+      const uniqueOptions = []
       const variantsLength = this.layer.variants.length
+
       if (!variantsLength || variantsLength === 1) {
         return []
       }
 
-      return this.layer.variants.map((variant, index) => ({
-        value: Number(index),
+      this.layer.variants.map(variant => (
+        Object.keys(variant.properties).find((key) => {
+          const keyIsAllowed = this.variantFilterProperties.includes(key)
+
+          if (!keyIsAllowed) { return false }
+
+          const existingOptions = uniqueOptions.find(field => field.title === key)
+
+          if (!existingOptions) {
+            uniqueOptions.push({ title: key, options: [variant.properties[key]] })
```

Comment:
A side effect in an array method other than `forEach` something we should keep away from